CompileIO/compile.io
# 113 limit submission attempts

# Addressed Issue(s)

#113 

# Goals

The ability to limit student submission attempts on a particular assignment

# Special Considerations

Need to have mongo running on your local machine
If Docker doesn't work for your computer you might not be able to test this

# Acceptance Criteria

If the assignment has a certain number of tries allowed the student is only able to submit that many times. If you update the assignment tries the student's ability to submit should also change accordingly.

# Testing

- Have front end and back end running
- Make a new course
- Make a new section
- Make a new Assignment
   - In the assignment make the number of tries 1
- reload the page after you make a new assignment (if the program doesn't already)

-Follow the links to your assignment again, you should be on the run tab for your assignment, submit a file, and run the code.

- Once your code is run you should see on the page "Reached Maximum Submission Attempts" since for your assignment the number of tries is 1

- Lastly, update your assignment so that the number of tries is 2
- reload the page after you update the assignment (if the program doesn't already)

- Go back to the code, and you should be able to run the code once more, and when you run it again you should see the same "Reached Maximum Submission Attempts" prompt

If all of this goes well this issue should be ready to receive an approving review



# Branch

113-Limit-Submission-Attempts
